CrossrefReferenceLinkingPlugin.php

```diff
@@ -0,0 +1,474 @@
+			if (!$doi) return false;
+			$publications = Repo::publication()->getCollector()->filterByDoiIds([$doi->getId()])->getMany();
+			if ($publications->count() < 1) return false;
+			$submission = Repo::submission()->get($publications->first()->getData('submissionId'));
+			$articleCitations = $citationDao->getByPublicationId($submission->getCurrentPublication()->getId());
+			$articleCitationsArray = $articleCitations->toArray();
+			if (!empty($articleCitationsArray)) {
+				$citationListNode = $preliminaryOutput->createElementNS($rfNamespace, 'citation_list');
+				foreach($articleCitationsArray as $citation) {
+					$rawCitation = $citation->getRawCitation();
+					if (!empty($rawCitation)) {
+						$citationNode = $preliminaryOutput->createElementNS($rfNamespace, 'citation');
+						$citationNode->setAttribute('key', $citation->getId());
+						// if Crossref DOI already exists for this citation, include it
+						// else include unstructred raw citation
+						if ($citation->getData($this->getCitationDoiSettingName())) {
+							$citationNode->appendChild($node = $preliminaryOutput->createElementNS($rfNamespace, 'doi', htmlspecialchars($citation->getData($this->getCitationDoiSettingName()), ENT_COMPAT, 'UTF-8')));
+						} else {
+							$citationNode->appendChild($node = $preliminaryOutput->createElementNS($rfNamespace, 'unstructured_citation', htmlspecialchars($rawCitation, ENT_COMPAT, 'UTF-8')));
+						}
+						$citationListNode->appendChild($citationNode);
+					}
+				}
+				$doiDataNode->parentNode->insertBefore($citationListNode, $doiDataNode->nextSibling);
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * During the article DOI registration with Crossref, get the citations diagnostic ID from the Crossref response.
+	 *
+	 * @param $hookName string Hook name 'crossrefexportplugin::deposited'
+	 * @param $params array [
+	 *  @option CrossrefExportPlugin
+	 *  @option string XML reposonse from Crossref deposit
+	 *  @option Submission
+	 * ]
+	 */
+	public function getCitationsDiagnosticId(string $hookName, array $params): bool
+	{
+		$response = & $params[1];
+		$submission = & $params[2];
+		// Get DOMDocument from the response XML string
+		$xmlDoc = new DOMDocument();
+		$xmlDoc->loadXML($response);
+		if ($xmlDoc->getElementsByTagName('citations_diagnostic')->length > 0) {
+			$citationsDiagnosticNode = $xmlDoc->getElementsByTagName('citations_diagnostic')->item(0); /** @var DOMNodeList $citationsDiagnosticNode */
+			$citationsDiagnosticCode = $citationsDiagnosticNode->getAttribute('deferred') ;
+			//set the citations diagnostic code and the setting for the automatic check
+			$submission->setData($this->getCitationsDiagnosticIdSettingName(), $citationsDiagnosticCode);
+			$submission->setData($this->getAutoCheckSettingName(), true);
+			$submission = Repo::submission()->edit($submission, array(), Application::get()->getRequest());
+		}
+		return false;
+	}
+
+	/**
```

Comment:
There must be some tabs here, I think the code formatter might be able to handle these as well.